quickwit-cluster/src/cluster.rs

```diff
@@ -75,30 +87,64 @@ impl Member {
     }
 }
 
-impl TryFrom<NodeId> for Member {
-    type Error = anyhow::Error;
+impl From<Member> for NodeId {
+    fn from(member: Member) -> Self {
+        Self::new(member.internal_id(), member.gossip_public_address)
+    }
+}
 
-    fn try_from(node_id: NodeId) -> Result<Self, Self::Error> {
-        let (node_unique_id_str, generation_str) = node_id.id.split_once('/').ok_or_else(|| {
+// Build a cluster member from [`ClusterStateSnapshot`].
+// TODO: instead of passing a [`ClusterStateSnapshot`], passing a `NodeState` would be nicer but
+// the struct is not yet public.
+fn build_member_from_chichat<'a>(
+    node_id: &'a NodeId,
+    cluster_state_snapshot: &'a ClusterStateSnapshot,
+) -> anyhow::Result<Member> {
+    let (node_unique_id_str, generation_str) = node_id.id.split_once('/').ok_or_else(|| {
+        anyhow::anyhow!(
+            "Could not create a Member instance from NodeId `{}`",
+            node_id.id
+        )
+    })?;
+    let node_state = cluster_state_snapshot
+        .node_states
+        .get(&node_id.id)
+        .ok_or_else(|| {
             anyhow::anyhow!(
-                "Could not create a Member instance from NodeId `{}`",
-                node_id.id
+                "Could not find node id `{}` in ChitChat state.",
+                node_unique_id_str,
             )
         })?;
-
-        Ok(Self {
-            node_unique_id: node_unique_id_str.to_string(),
-            generation: generation_str.parse()?,
-            gossip_public_address: node_id.gossip_public_address,
-            is_self: false,
+    let available_services = node_state
```

Comment:
`NodeState` should have an accessor for getting the value of a key.

Reply:
NodeState has already an accessor, it will eventually store more (key, string value) pairs and we will need to parse more values. And I don't see how to make this part prettier :/.

Reply:
I think we just need wrappers:
```rust
struct QuikwitNode(NodeState)

impl QuickwitNode {
  // Accessors here
} 

impl From<NodeState> for QuickwitNode {
...
}
```
We can also wrap `ClusterStateSnapshot` if necessary.

Reply:
Once we have a public `NodeState` we can add the wrapper.